### ISSUE TITLE
Loss of dimension bug in numpy.ma.__getitem__ for arrays of shape (n, 1) or (1, n) or (n, 1, 1, ...) (migrated from Trac #744)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#1351]
Original ticket http://projects.scipy.org/numpy/ticket/744
Reported 2008-04-19 by trac user dharland, assigned to atmention:pierregm.

There is a bug in MaskedArray.**getitem** that causes a loss in dimension for masked arrays of shape (n, 1) or (1, n) or more generally, any shape of the form (1,)_a + (n,) + (1,)_b

Heres an example

```
  # A 2d column vector
  x = ma.arange(5).reshape( (-1, 1) )

  # mask a point
  x[3] = ma.masked

  # Different behavior for unmasked and masked items
  print x[0], x[0].shape
  print x[3], x[3].shape
```

this produces as output

```
  [0] (1,)    # Correct
  -- ()       # This should be [--]
```

As a result code cannot reliably iterate over rows of a 2d array (in case its in fact a (n,1) array)

The error appears in the following code snippet in numpy/core/ma.py:
            mi = m[i]
            if mi.size == 1:    # <-- Should this be if mi.size == 1 and mi.ndim == 0
                if mi:
                    return masked    # <-- dimension loss here if mi.ndim > 0
                else:
                    return dout

**Comment by thouis** [comment]
Comment in Trac by atmention:jarrodmillman, 2008-04-28

**Comment by thouis** [comment]
Comment in Trac by atmention:pierregm, 2008-04-29

What version are you using ? It seems to work OK with v4807

```
>>>x=ma.arange(5).reshape((-1,1))
>>>x[3]=ma.masked
>>>print x[0],x[0].shape
[0] (1,)
>>>print x[3],x[3].shape
[--] (1,)
```

**Comment by thouis** [comment]
Comment in Trac by trac user dharland, 2008-04-29

Replying to [comment:2 pierregm]:

> What version are you using ? It seems to work OK with v4807

I'm using release 1.0.4.  I see that MaskedArray.**getitem** on trunk has changed substantially versus this last release and looks fixed, so I guess you can close this.

Is there a timeline for the next point release?

**Comment by thouis** [comment]
Comment in Trac by atmention:jarrodmillman, 2008-04-29

**Comment by thouis** [comment]
Comment in Trac by atmention:jarrodmillman, 2008-04-29

**Comment by thouis** [comment]
Comment in Trac by atmention:pierregm, 2008-04-29

Let's consider it closed (or better still, invalid) for 1.1, then.
